src/containers/HomeContainer/Home.js

```diff
@@ -0,0 +1,171 @@
+  const universityId = currentUser.universityId
+
+  /**
+   * Helper function to filter courses by university id
+   * @param offerings Pass in all offerings that student is allowed to access only
+   */
+  const filterCourses = (offerings) => {
+    const newOfferings = []
+    for (const i in offerings) {
+      if (offerings[i].term.universityId === universityId) {
+        newOfferings.push(offerings[i])
+      }
+    }
+
+    return newOfferings
+  }
+
+  const [courses, setCourses] = useState([])
+  useEffect(() => {
+    const fetchCourseInfo = async () => {
+      const offerings = await getOfferingsData()
+      const studentCourses = filterCourses(offerings)
+      setCourses(studentCourses)
+    }
+    fetchCourseInfo()
+  }, [setCourses])
+
+  const onCourseSelected = (courseId) => {
+    navigation.push(STACK_SCREENS.COURSE_PLAYLISTS, { courseId })
+  }
+
+  /**
+   * Renders a single course in the course list.
+   * @param {Object} item The underlying data for the item to render.
+   */
+  const renderCourseItem = ({ item }) => {
+    if (item.length === 0) return null
+
+    const course = item.courses[0]
+    const { courseName } = item.offering
+    const courseDescription = item.offering.description
+    const courseId = item.offering.id
+
+    return (
+      <View style={styles.container} key={courseId}>
+        <TouchableNativeFeedback onPress={() => onCourseSelected(courseId)}>
+          <View style={styles.cardContainer}>
+            <CourseCard
+              departmentAcronym={course.departmentAcronym}
+              courseNumber={course.courseNumber}
+              courseName={courseName}
+              courseDescription={courseDescription}
+            />
+          </View>
+        </TouchableNativeFeedback>
+        <View key={courseName} style={styles.recContainer}>
+          <Recommend key={courseId} navigation={navigation} courseId={courseId} mode={mode} />
+        </View>
+      </View>
+    )
+  }
+
+  /**
+   * Render universities' course offerings in a dropdown picker based on university id.
+   */
+  const renderUniversityDropDown = () => {
+    const [universities, setAllUniversities] = useState([])
+    useEffect(() => {
+      const fetchUniversities = async () => {
+        const allUnis = await getUniversities()
+        setAllUniversities(allUnis)
+      }
+      fetchUniversities()
+    }, [setAllUniversities])
+
+    const universityItems = universities.map((uni) => {
+      return <Picker.Item accessibilityRole="button" key={uni.id} value={uni.id} label={uni.name} />
+    })
+
+    const [university, setUniversity] = useState(universityId)
+
+    const onUniversitySelected = async (newUniversityId) => {
+      // Reload the page for the selected universityId
+      setUniversity(newUniversityId)
+
+      const newUnicourses = await getOfferingsData()
+      const newCourses = filterCourses(newUnicourses)
+
+      try {
+        setCourses(newCourses)
+        renderCourseItem()
```

Comment:
Why are you calling renderCourseItem? Is the intent to rerender the screen? If so, you don't need to call that here, React should automatically rerender when the state is changed. When I run the app on my phone, it has some really weird behavior and it says it has a memory leak. This could be partially why.